parsing/tests/parsing.spec.ts

```diff
@@ -75,6 +75,10 @@ describe('common', () => {
                 for (const product of recipe.products) {
                     parts.add(product.part);
                 }
+                if (recipe.products.length === 0) {
+                    console.log('No products for recipe:', recipe.id);
+                    expect('Recipe ' + recipe.id + ' has no products').toBe('');
```

Comment:
Sorry this one doesn't match a ton of sense, you've got an `expect` which is asserting a string to be `''`. It's not actually testing anything?

Reply:
All it does is expects something in the case of recipe.products.length is 0.

It would be better as:

`expect(recipe.products.length).not.toBe(0)`

Reply:
It's more an assert. The thinking here is if there are no products, it throws an error with the description of the recipe that has the issue, so that it's easier to resolve. If I just  `expect(recipe.products.length).not.toBe(0)`, I have no idea what recipe failed and I have to start a whole new line of debugging. 

Does that make sense, or still sound crazy?

Reply:
If you change `console.log` to `console.error` and **not** use `--silent` on the test run it'll show up the console in the output.

Reply:
That is another solution. I left the `--silent` on because you started with it (I usually prefer verbose for my tests). I'm flexible here. Whatever you think is best. :)

Reply:
Well what we can do is make it fail, and if it's failing we can run the tests without --silent and see why, best of both words :)

Reply:
I'm not clear what you need to close this out. Please be more specific and I'll do it! :)

Reply:
> `expect(recipe.products.length).not.toBe(0)`

This, we need the test to fail, you can keep the logging but do it as console.errors